Unable to create GLSL compute shader on intel integrated graphics

Hi

I'm attempting to install Render Pipeline on an integrated intel gpu - Intel Ivybridge Mobile.

```
-------------------------------------------------------------------------------`

Render Pipeline Setup 1.3

-------------------------------------------------------------------------------


[ 01 ]  Checking Panda3D Modules


[ 02 ]  Checking requirements ..
	Running script: data/setup/check_requirements.py


The C++ modules of the pipeline are faster and produce better 
results, but we will have to compile them. As alternative, 
a Python fallback is used, which is slower and produces worse 
results. Also some plugins only partially work with the python 
fallback (e.g. PSSM). Do you want to compile the C++ modules? (y/n): y


[ 03 ]  Downloading the module builder ...
	Running script: rpcore/native/update_module_builder.py
	Downloading:   0.00  B/s |                               |  0%   ETA:  --:--:-- Downloading:  92.82 GB/s |################################|100%   Time: 0:00:00


[ 04 ]  Building the native code .. (This might take a while!)
	Running script: rpcore/native/build.py


[ 05 ]  Generating .txo files ...
	Running script: data/generate_txo_files.py


[ 06 ]  Filtering default cubemap ..
	Running script: data/default_cubemap/filter.py
:display:gsg:glgsg(error): Could not create a GLSL compute shader.
Assertion failed: do_has_ram_mipmap_image(cdata, n) at line 4948 of panda/src/gobj/texture.cxx
Traceback (most recent call last):
  File "filter.py", line 105, in <module>
    Application()
  File "filter.py", line 93, in __init__
    dest_cubemap.write(join(filter_dir, "{}-#.png".format(mipmap)), 0, 0, True, False)
AssertionError: do_has_ram_mipmap_image(cdata, n) at line 4948 of panda/src/gobj/texture.cxx
Failed to execute 'filter.py'
Output: Command '['/usr/bin/python', '-B', 'filter.py']' returned non-zero exit status 1 
 Filtering mipmap 0
 Extracting data ..
 Writing data ..




It seems the setup failed, do you want to open the troubleshooting page for this step? (y/n):  n

Setup failed:	 Python script didn't return properly!

Please fix the above errors and then restart the setup.
```

I've tried reinstalling drivers, updating and removing them, to no avail. 
I'm using Panda 1.10 with the latest version of render pipeline.

Any idea what the issue is, any way to solve it?
_Thanks!_

yes Linux..

I installed the mesa 17.0.2 drivers which supposedly have support for OpenGL 4.5.
Unfortunately it turns out that isn't really the case.

Slightly different error running the script again:
```
[ 06 ]  Filtering default cubemap ..
	Running script: data/default_cubemap/filter.py
:display:gsg:glgsg(error): An error occurred while compiling GLSL compute shader filter.compute.glsl:
filter.compute.glsl:1(10): error: GLSL 4.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.00 ES, and 3.00 ES
filter.compute.glsl:0(0): error: Compute shaders require GLSL 4.30 or GLSL ES 3.10
Assertion failed: do_has_ram_mipmap_image(cdata, n) at line 4948 of panda/src/gobj/texture.cxx
Traceback (most recent call last):
  File "filter.py", line 105, in <module>
    Application()
  File "filter.py", line 93, in __init__
    dest_cubemap.write(join(filter_dir, "{}-#.png".format(mipmap)), 0, 0, True, False)
AssertionError: do_has_ram_mipmap_image(cdata, n) at line 4948 of panda/src/gobj/texture.cxx
Failed to execute 'filter.py'
```

--- a/config/panda3d-config.prc
+++ b/config/panda3d-config.prc
@@ -103,6 +103,9 @@ gl-force-fbo-color #f
 
 # ----------- OpenGL / Performance Settings ------------
 
+# Require OpenGL 4.3 at least, necessary for Intel drivers on Linux
+gl-version 4 3
+
 # Animations on the gpu. The default shader has to get adjusted to support this
 # feature before this option can be turned on.
 # hardware-animated-vertices #t

--- a/data/default_cubemap/filter.py
+++ b/data/default_cubemap/filter.py
@@ -42,6 +42,7 @@ class Application(ShowBase):
             gl-coordinate-system default
             notify-level-display error
             print-pipe-types #f
+            gl-version 4 3
         """)
 
         ShowBase.__init__(self)

--- a/data/film_grain/generate.py
+++ b/data/film_grain/generate.py
@@ -48,6 +48,7 @@ class Application(ShowBase):
             gl-coordinate-system default
             notify-level-display error
             print-pipe-types #f
+            gl-version 4 3
         """)
 
         ShowBase.__init__(self)

--- a/rpplugins/clouds/resources/precompute.py
+++ b/rpplugins/clouds/resources/precompute.py
@@ -41,6 +41,7 @@ class Application(ShowBase):
             gl-coordinate-system default
             notify-level-display error
             print-pipe-types #f
+            gl-version 4 3
         """)
 
         ShowBase.__init__(self)

